edx_notifications/data.py

```diff
@@ -0,0 +1,172 @@
+"""
+All pure data objects that the various Data Providers will product. This is help avoid
+generic dictionaries from being passed around, plus this will help avoid any
+implicit database-specific bindings that come with any uses of ORMs.
+"""
+
+from django.core.exceptions import ValidationError
+
+from edx_notifications import const
+
+from edx_notifications.base_data import (
+    StringField,
+    DictField,
+    DateTimeField,
+    IntegerField,
+    EnumField,
+    RelatedObjectField,
+    BaseDataObject
+)
+
+
+class NotificationChannel(BaseDataObject):
+    """
+    Specifies a channel through which a notification is delivered
+    """
+
+    # the internal name of the channel
+    name = StringField()
+
+    # the human readible name of the channel
+    display_name = StringField()
+
+    # the human readible description of the channel
+    description_name = StringField()
+
+
+class NotificationType(BaseDataObject):
+    """
+    The Data Object representing the NotificationType
+    """
+
+    # the name (including namespace) of the notification, e.g. open-edx.lms.forums.reply-to-post
+    name = StringField()
+
+    # the human readible string of the name of the notification
+    display_name = StringField()
+
+    # the human readible string that describes the notification
+    display_description = StringField()
+
+    # default delivery channel for this type
+    # None = no default
+    default_channel = RelatedObjectField(NotificationChannel)
```

Comment:
I'm debating whether to change this to an in-memory for now since MVP will be running in-proc. I think one of the original appeal to have this go into SQL was to have a ForeignKey constraint on the type_name.